File: Domain/Payoff.cs
namespace Domain
{
    public class Payoff
    {
        public int Id { get; set; }
        public int LoanId { get; set; }
        public decimal Amount { get; set; }
        public int AccountId { get; set; }
        public DateTime Date { get; set; } = DateTime.UtcNow.Date;
        public string Description { get; set; }
        public int CurrencyId { get; set; }

        public virtual Currency Currency { get; set; }
        public virtual Loan Loan { get; set; }
        public virtual Account Account { get; set; }
    }
}

File: client-app/src/app/models/Payoff.ts
import dayjs from "dayjs";
import { LoanType } from "./enums/LoanType";

export interface Payoff {
    id: number;
    loanId: number;
    amount: number;
    accountId: number;
    date: Date;
    description?: string;
    currencyId: number;
}

export interface PayoffCreateValues {
    amount: number | null;
    accountId: number | string;
    date: dayjs.Dayjs | Date;
    description?: string;
}

export interface CollectivePayoffValues {
    amount: number | null;
    accountId: number | string;
    date: dayjs.Dayjs | Date;
    loanType: LoanType | string;
}